app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -488,8 +497,22 @@ def parse_args(args)
 
       args.except!(:hardware_version) if args[:hardware_version] == 'Default'
 
-      firmware_type = args.delete(:firmware_type)
-      args[:firmware] = firmware_mapping(firmware_type) if args[:firmware] == 'automatic'
+      firmware_type = args.delete(:firmware_type).to_s
+
+      if args[:firmware] == 'automatic'
+        if firmware_type == 'none' || firmware_type.empty?
+          args[:firmware] = 'bios'
+        else
+          args[:firmware] = firmware_type
+        end
+      end
+
+      if args[:firmware]&.start_with?('uefi')
```

Comment:
For me, this is pretty hard to understand why the firmware type is changed from uefi to efi. Maybe a comment why this needs to be done would help for the furture.
Additionally, I think the code can be simplified:

`
args[:firmware] = 'efi' if args[:firmware]&.start_with?('uefi')
args[:secure_boot] = args[:firmware] == 'uefi_secure_boot'
`

Reply:
I think that comment should describe the block above (automatic firmware) as well. The handling as a whole is not intuitive so a comment can make sense.